src/components/editor/SinopiaResourceTemplates.jsx

```diff
@@ -23,63 +23,69 @@ class SinopiaResourceTemplates extends Component {
   }
 
   async componentDidUpdate(prevProps) {
-    if (this.props.updateKey > prevProps.updateKey) {
-      const groupPromise = getGroups().then(data => {
-        return data
-      }).catch(() => {})
+    if (this.props.updateKey <= prevProps.updateKey)
+      return
 
-      await this.fulfillGroupPromise(groupPromise).then(async () => {
-        this.state.groupData.map(group => {
+    getGroups()
+      .then(data => {
+        const contains = [].concat(data.response.body.contains)
+        this.setState({groupData: contains})
+      })
+      .catch(error => {
+        this.setState({message: error})
+      })
+      .then(() => {
+        return this.state.groupData.map(group => {
           const groupName = this.resourceToName(group)
-          listResourcesInGroupContainer(groupName).then((data) => {
-            this.fulfillGroupData(data)
-          }).catch(() => {})
+          return listResourcesInGroupContainer(groupName)
+        })
+      })
+      .then(groupResources => {
+        Promise.all(groupResources).then(responses => {
+          responses.forEach(group => {
+            this.handleGroupResponse(group)
+          })
         })
       })
-    }
   }
 
-  fulfillGroupPromise = (promise) => {
-    promise.then(data => {
-      const contains = [].concat(data.response.body.contains)
-      this.setState({groupData: contains})
-    }).catch((error) => {
-      this.setState({message: error})
-    })
-    return promise
-  }
+  handleGroupResponse = data => {
+    if (data.response.body === null)
+      return
 
-  fulfillGroupData = (data) => {
     const groupName = this.resourceToName(data.response.body['@id'])
+    // Array-ify the value of `contains` and filter out falsy values like null and undefined
+    const contains = [].concat(data.response.body.contains).filter(Boolean)
+
+    contains.forEach(child => {
```

Comment:
comment from discussion:   `forEach` for iterating thru without needing the result after;  `map` when you need the result what happens for each step.

There may be some foo around `forEach` vs `map`

Reply:
caution:  don't use "return" with forEach or you'll be surprised!

Reply:
`return` within `.forEach()` functions like `next` in Ruby or `continue` in Java—it merely terminates the current iteration. It does not return a value to the containing context.